[ICDS Dashboard]Changing Height eligibility from 6 months - 5 years to 0-5 years

ShubhamBansal1997 opened the pull request:
<!--- Provide a link to the ticket or document which prompted this change -->

##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
Identifying the areas in the Dashboard where the height eligiblity is 6mnth to 5 years and not 0-5 years

https://app.asana.com/0/0/1153564689437801/f

##### FEATURE FLAG
<!--- If this is specific to a feature flag, which one? -->

##### PRODUCT DESCRIPTION
<!--- For non-invisible changes, describe user-facing effects. -->

ShubhamBansal1997 committed:
[fix]: change the default age filter to 0-5 years from 6 months-5 years
for age filter which is used in the height fields

- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

ShubhamBansal1997 committed:
[fix]: reduce the age from 6 years to 5 year in the AWC Report > child_beneficiary_list

- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

Rohit25negi commented on custom/icds_reports/reports/awc_reports.py:
```diff
@@ -335,7 +335,7 @@ def get_awc_reports_pse(config, month, domain, show_test=False):
 def get_awc_reports_maternal_child(domain, config, month, prev_month, show_test=False, icds_feature_flag=False):
 
     def get_data_for(date):
-        age_filters = {'age_tranche': 72} if icds_feature_flag else {'age_tranche__in': [0, 6, 72]}
```
Hey, Did you check below how this age_filters is being used?. 
I think its being used like `select all from table excluding age_tranche=60` that would include all children accept children of age 5 years.

Rohit25negi replied:
Since we need to include children less than 72 month of age. You could reuse the `include_record_by_age_column` function created in this PR https://github.com/dimagi/commcare-hq/pull/26125/files and work this out similarly.

Rohit25negi commented on custom/icds_reports/reports/awc_reports.py:
```diff
@@ -1012,7 +1012,7 @@ def get_awc_report_beneficiary(start, length, draw, order, filters, month, two_b
     filters['open_in_month'] = 1
     filters['valid_in_month'] = 1
     if filters.get('age_in_months__range') is None:
-        filters['age_in_months__lte'] = 72
```
I think this does not need to change. This child beneficiary list intentionally show list of all children  0-6years.

Rohit25negi replied:
ping!!

Rohit25negi reviewed (changes requested):
Left a few comments. There are some logical problems.

ShubhamBansal1997 committed:
[fix]: used include_record_by_age_column and fix basic tests

- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

Rohit25negi commented on custom/icds_reports/reports/awc_reports.py:
```diff
@@ -345,12 +345,12 @@ def get_data_for(date):
         )
-        wasting_moderate = exclude_records_by_age_for_column(
-            age_filters,
+        wasting_moderate = include_records_by_age_for_column(
```
hey, could you please try debugging this function locally. What query is being prepared here?
As this looks `{'age_tranche__lt': 60}`. Will exclude kids of age 5 years. as this will become `<60`

ShubhamBansal1997 committed:
[fix]: remove unwanted feature flag and also updated age filters to age
less than equal to 60 months or 5 years

- Github Issue:

Authored-by: Shubham Bansal <shubhambansal17@hotmail.com>
Signed-off-by: Shubham Bansal <shubhambansal17@hotmail.com>

Rohit25negi reviewed (approved):
Looks good thanks.
